change priorities of nest eradication and nest spawning

Blue-Award opened the pull request:
fixes https://github.com/freeorion/freeorion/issues/4952

maybe the effect stacking was to prevent monster spawning when nest is eradicated, but in practice I see nest eradication prevented when monster spawns.

Removing stacking group was easiest fix for me though it may mean that monster spawns as nest is being eradicated, by I suppose that's better than nest not being eradicated and losing production on the building

I am open to suggestions what other stacking group it could be so that eradication triggers first and monster spawning doesn't trigger later, if that's possible. Ultimately I don't mind though if monster spawns but nest gets removed

agrrr3 reviewed (changes requested):
some files include the priorities.macros without actually using it. please remove those includes.
i think the include belongs into the monster_nest.macros
else looks good to mie

Blue-Award committed:
put eradicator's effect on higher priority than monster spawn

so eradicator works properly in case monster would spawn - and monster actually doesn't spawn, it gets prevented

agrrr3 reviewed (approved):
LGTM
